src/Tizen.NUI.Gadget/Tizen.NUI/OneShotService.cs

```diff
@@ -0,0 +1,206 @@
+            }
+            TizenCore.Shutdown();
+        }
+
+        /// <summary>
+        /// Override this method to define the behavior when the OneShotService is created.
+        /// Calling 'base.OnCreate()' is necessary in order to
+        /// emit the 'OneShotServiceLifecycleState.Created` state
+        /// </summary>
+        /// <since_tizen> 13 </since_tizen>
+        protected virtual void OnCreate()
+        {
+            Log.Info("[OnCreate]");
+            if (State == OneShotServiceLifecycleState.Initialized)
+            {
+                State = OneShotServiceLifecycleState.Created;
+                NotifyLifecycleChanged();
+            }
+        }
+
+        /// <summary>
+        /// Override this method to define the behavior when the OneShotService is destroyed.
+        /// Calling 'base.OnDestroyed()' is necessary in order to
+        /// emit the 'OneShotServiceLifecycleState.Destroyed` state
+        /// </summary>
+        /// <since_tizen> 13 </since_tizen>
+        protected virtual void OnDestroy()
+        {
+            Log.Info("[OnDestroy]");
+            if (State == OneShotServiceLifecycleState.Running)
+            {
+                State = OneShotServiceLifecycleState.Destroyed;
+                NotifyLifecycleChanged();
+            }
+        }
+
+        /// <summary>
+        /// Starts the OneShotService execution on a worker thread.
+        /// Initializes the service task and triggers the OnCreate event.
+        /// If AutoClose is enabled, automatically destroys the service after execution.
+        /// </summary>
+        /// <since_tizen> 13 </since_tizen>
+        public void Run()
+        {
+            Log.Warn($"Name = {Name}");
+            if(_task != null && _task.Running)
+            {
+                Log.Info($"{Name} is already running");
+                return;
+            }
+
+            if (State == OneShotServiceLifecycleState.Initialized)
+            {
+                State = OneShotServiceLifecycleState.Running;
+                NotifyLifecycleChanged();
+
+                _task = TizenCore.Spawn(Name);
+                _task.Post(() => { OnCreate(); });
+
+                if (AutoClose)
+                {
+                    _task.Post(() =>
+                    {
+                        OnDestroy();
+                        _task.Quit();
+                    });
+                }
+            }
+        }
+
+        /// <summary>
+        /// Stops the OneShotService execution and releases resources.
+        /// </summary>
+        /// <param name="waitForJoin">Whether to wait for complete service termination</param>
+        /// <since_tizen> 13 </since_tizen>
+        public void Quit(bool waitForJoin)
+        {
+            Log.Warn($"Name = {Name}");
+            if(_task == null || State == OneShotServiceLifecycleState.Destroyed)
+            {
+                Log.Info($"{Name} was already destroyed");
+                return;
+            }
```

Comment:
Please check the waitForJoin and modify this implementation.
If the waitForJoin is 'true' and the _task is not null, we should call _task.Dispose() method to wait until the service is terminated properly.

Reply:
Modify on this [commit](https://github.com/Samsung/TizenFX/pull/7211/commits/211454f641d4346d6827d12e77111cd51f4d92ea)

Please review this code :)